core/src/main/java/org/jdbi/v3/core/statement/StatementBuilder.java

```diff
@@ -57,29 +57,56 @@ public interface StatementBuilder {
      *
      * @param conn the JDBC Connection the statement is being created for
      * @param sql the translated SQL which should be prepared
-     * @param ctx Statement context associated with the SqlStatement this is building for
+     * @param ctx Statement context associated with the SqlStatement
      *
      * @return a CallableStatement for the given arguments
      *
      * @throws SQLException if anything goes wrong preparing the statement
      */
     CallableStatement createCall(Connection conn, String sql, StatementContext ctx) throws SQLException;
 
+    /**
+     * Called to close an individual statement created from this builder.
+     *
+     * @param conn the connection to close.
+     * @param sql  the translated SQL which was prepared.
+     * @param stmt the statement.
+     * @throws SQLException if anything goes wrong closing the statement.
+     * @deprecated Use either {@link #close(PreparedStatement, Connection, String, StatementContext)} to close a {@link PreparedStatement} or
+     * {@link #close(Statement, Connection, StatementContext)} to close a {@link Statement} for a batch.
+     */
+    @Deprecated
+    void close(Connection conn, String sql, Statement stmt) throws SQLException;
+
     /**
      * Called to close an individual prepared statement created from this builder.
      *
-     * @param conn the connection to close
-     * @param sql the translated SQL which was prepared
-     * @param stmt the statement
+     * @param stmt the prepared statement.
+     * @param conn the connection to close.
+     * @param sql  the translated SQL which was prepared.
+     * @param ctx Statement context associated with the SqlStatement.
+     * @throws SQLException if anything goes wrong closing the statement.
+     */
+    default void close(PreparedStatement stmt, Connection conn, String sql, StatementContext ctx) throws SQLException {
+        close(conn, sql, stmt);
+    }
+
+    /**
+     * Called to close an individual statement created from this builder.
      *
-     * @throws SQLException if anything goes wrong closing the statement
+     * @param stmt the prepared statement.
+     * @param conn the connection to close.
+     * @param ctx Statement context associated with the SqlStatement.
+     * @throws SQLException if anything goes wrong closing the statement.
      */
-    void close(Connection conn, String sql, Statement stmt) throws SQLException;
+    default void close(Statement stmt, Connection conn, StatementContext ctx) throws SQLException {
+        close(conn, "", stmt);
+    }
 
     /**
      * Called when the handle this StatementBuilder is attached to is closed.
      *
      * @param conn the connection to close.
```

Comment:
This doc (not part of your immediate change) is slightly misleading, since you are not supposed to close the connection here.

Reply:
ok, not sure I understand that comment. :-)

Reply:
I think I understood it later. I changed the wording of the javadoc.